### PR TITLE
Add FocusZone Prop for Tab Navigation Abstraction

### PR DESCRIPTION
**Pull request by samuelfreiberg**
### Platforms Impacted
- [ ] iOS
- [ ] macOS
- [X] win32 (Office)
- [X] windows
- [ ] android

### Description of changes

This PR adds a prop on FocusZone, tabKeyNavigation, that allows a user to abstract the "Tab" key when pressed within the FZ.

The default "Tab" behavior when pressing within a FocusZone has been to move focus out of the FZ. With this PR, we add more abstraction and allow users to change the behavior. Here are all the "Tab" navigation options:

None - Tab navigates you out of the FocusZone. This is Default.
NavigateWrap - Navigate the FZ with Tab. Circular navigation at ends.
NavigateStopAtEnds - Navigate the FZ with Tab. Stop navigation at ends.
Normal - Navigate the FZ with Tab. Don't trap focus, tabbing at ends moves you out.

I can't add E2E tests yet because the native change isn't merged yet either. I will after!

### Verification

Verified in the test app

| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
| Screenshot or description before this change | Screenshot or description with this change |

### Pull request checklist

This PR has considered (when applicable):
- [ ] Automated Tests
- [ ] Documentation and examples
- [ ] Keyboard Accessibility
- [ ] Voiceover
- [ ] Internationalization and Right-to-left Layouts

### CODE REVIEW

File: packages/components/FocusZone/src/FocusZone.types.ts [hunk 1]
```diff
@@ -37,6 +37,27 @@ export type FocusZoneProps = React.PropsWithChildren<{
...
+   * "NavigateStopAtEnds" : Tab allowed within FZ and stops at ends
+   * "Normal" : Tab allowed within FZ, and focus is not trapped. Tabbing at the last element bring you out of the FZ.
+   *
+   * Recommended Usage - Moving focus OUT of the FocusZone - Custom Key Handlers
+   *      Example: Subscribe to key events. On the pressing of "Escape", move focus out of the FocusZone.
+   *
+   * Recommended Usage - Moving focus INTO the FocusZone
+   *    Let's take a CardList, for example. If a user is navigating through elements of the List, we don't want
+   *    focus to land within the FocusZone on initial focus. If it does, it will trap focus within the FocusZone and
+   *    ruin the user's workflow.
```
Comment (Saadnajmi) [inline packages/components/FocusZone/src/FocusZone.types.ts:55]: Interesting, I think I fixed a similar issue by making the FZ itself focusable and its' own default tubbable element on macOS, should the user wish 😅

File: packages/components/FocusZone/src/FocusZone.types.ts [hunk 2]
```diff
@@ -37,6 +37,27 @@ export type FocusZoneProps = React.PropsWithChildren<{
...
+   *    Instead, once focus lands on the parent of the FocusZone, implement a custom key handler that will enable
+   *    the FocusZone.
+   */
+  tabKeyNavigation?: FocusZoneTabNavigation;
```
Comment (Saadnajmi) [inline packages/components/FocusZone/src/FocusZone.types.ts:59]: Can you mark this as win32 only?

Comment (samuelfreiberg) [reply packages/components/FocusZone/src/FocusZone.types.ts:59]: Yup! Good call

Comment (samuelfreiberg) [reply packages/components/FocusZone/src/FocusZone.types.ts:60]: Updated in newest iteration

File: packages/components/FocusZone/src/FocusZone.types.ts (hunk 1 above)
Comment (samuelfreiberg) [reply packages/components/FocusZone/src/FocusZone.types.ts:56]: Interesting - did you add a View as a slot to the FocusZone?

File: apps/fluent-tester/src/TestComponents/FocusZone/FocusZoneE2ETest.tsx [hunk 3]
```diff
@@ -25,6 +25,7 @@ import { commonTestStyles } from '../Common/styles';
 import { testProps } from '../Common/TestProps';
 
 export const FocusZoneDirections: FocusZoneDirection[] = ['bidirectional', 'horizontal', 'vertical', 'none'];
+export const FocusZoneTabNavigations: FocusZoneTabNavigation[] = ['None', 'NavigateWrap', 'NavigateStopAtEnds', 'Normal'];
```
Comment (PPatBoyd) [inline apps/fluent-tester/src/TestComponents/FocusZone/FocusZoneE2ETest.tsx:28]: nit created and exported from the E2E test to the non-E2E test page because we expect to use this enum to write E2E tests, or ?

File: packages/components/FocusZone/src/FocusZone.ts [hunk 4]
```diff
@@ -15,7 +15,7 @@ const filterOutComponentRef = (propName) => propName !== 'componentRef';
 
 export const FocusZone = composable<FocusZoneType>({
   usePrepareProps: (userProps: FocusZoneProps, useStyling: IUseStyling<FocusZoneType>) => {
-    const { componentRef, defaultTabbableElement, isCircularNavigation, ...rest } = userProps;
+    const { componentRef, defaultTabbableElement, isCircularNavigation, tabKeyNavigation, ...rest } = userProps;
```
Comment (PPatBoyd) [inline packages/components/FocusZone/src/FocusZone.ts:18]: looks like it could be left in `...rest`?

File: packages/components/FocusZone/src/FocusZone.types.ts [hunk 5]
```diff
@@ -45,6 +67,7 @@ export type FocusZoneProps = React.PropsWithChildren<{
 
 export interface NativeProps extends Omit<FocusZoneProps, 'isCircularNavigation'> {
```
Comment (PPatBoyd) [inline packages/components/FocusZone/src/FocusZone.types.ts:68]: driveby nit, should `isCircularNavigation` be deprecated?

**Review by PPatBoyd — APPROVED** [review]

File: apps/fluent-tester/src/TestComponents/FocusZone/FocusZoneE2ETest.tsx (hunk 3 above)
Comment (samuelfreiberg) [reply apps/fluent-tester/src/TestComponents/FocusZone/FocusZoneE2ETest.tsx:28]: Ah, yes. Good catch. I had started writing E2E tests, until I realized we'd have to wait for the native changes. But that PR will come as soon as the code is checkpointed natively. I'll keep this here for now as it'll be used in the next PR :)

File: packages/components/FocusZone/src/FocusZone.ts (hunk 4 above)
Comment (samuelfreiberg) [reply packages/components/FocusZone/src/FocusZone.ts:18]: You're correct - changing in newest iteration

File: packages/components/FocusZone/src/FocusZone.types.ts (hunk 5 above)
Comment (samuelfreiberg) [reply packages/components/FocusZone/src/FocusZone.types.ts:68]: I don't believe so, it's still a heavily used abstraction for arrow keys. What's your thinking here?

Comment (samuelfreiberg) [reply packages/components/FocusZone/src/FocusZone.types.ts:68]: Will continue this offline
